lib/src/arithmetic/partial_product_generator.dart

```diff
@@ -0,0 +1,561 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// partial_product_generator.dart
+// Partial Product matrix generation from Booth recoded multiplicand
+//
+// 2024 May 15
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:math';
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// Methods for sign extending the [PartialProductGenerator]
+enum SignExtension {
```

Comment:
Upon rereading this, perhaps this is an opportunity to use a base class with sub-classes that extend it rather than an enum with a case statement?

Reply:
I created a set of subclasses, kept one in the main file and moved the others to a test file.   I use the simpler ones for debug to make sure I don't have a sign extension error that is blocking, or when doing an enhancement (like accumulate).